arangod/Cluster/UpdateCollection.cpp

```diff
@@ -77,21 +77,38 @@ UpdateCollection::UpdateCollection(MaintenanceFeature& feature, ActionDescriptio
   }
   TRI_ASSERT(desc.has(FOLLOWERS_TO_DROP));
 
+  TRI_ASSERT(desc.has(OLD_CURRENT_COUNTER));
+
   if (!error.str().empty()) {
-    LOG_TOPIC("a6e4c", ERR, Logger::MAINTENANCE) << "UpdateCollection: " << error.str();
+    LOG_TOPIC("a6e4c", ERR, Logger::MAINTENANCE)
+        << "UpdateCollection: " << error.str();
     _result.reset(TRI_ERROR_INTERNAL, error.str());
     setState(FAILED);
   }
 }
 
 void handleLeadership(LogicalCollection& collection, std::string const& localLeader,
-                      std::string const& plannedLeader, std::string const& followersToDrop) {
+                      std::string const& plannedLeader,
+                      std::string const& followersToDrop, std::string const& databaseName,
+                      uint64_t oldCounter, MaintenanceFeature& feature) {
   auto& followers = collection.followers();
 
   if (plannedLeader.empty()) {   // Planned to lead
     if (!localLeader.empty()) {  // We were not leader, assume leadership
       followers->setTheLeader(std::string());
       followers->clear();
+      // This will block the thread until we fetched a new current version
```

Comment:
```followers->setTheLeader(std::string());```
must be done after the `waitForLargerCurrentCounter`, and either atomically with the failover update or after failover candidates are updated.
```followers->clear();``` must be removed.

Reply:
Did atomic variant